doc/quick-start/README.md

```diff
@@ -55,16 +32,61 @@ docker exec node4 /bin/sh -c "/usr/local/bin/etcdctl --endpoints=\"http://172.20
 docker exec node4 /bin/sh -c "/usr/local/bin/etcdctl --endpoints=\"http://172.20.0.3:2379\" get A"
 ```
 
+## Validation Test
+
+```bash
+docker cp node1:/usr/local/bin/lock_client ./scripts
+
+./scripts/validation_test.sh
+```
+
 ## Benchmark
 
 ```bash
 ./scripts/benchmark.sh
 ```
 
-## Validation Test
+## Build from scratch
+
+### Install dependencies
+
+#### Ubuntu/Debian
 
 ```bash
-./scripts/validation_test.sh
+sudo apt-get install -y autoconf autogen libtool
+
+# requires protobuf-compiler >= 3.15
+git clone --branch v3.21.12  --recurse-submodules https://github.com/protocolbuffers/protobuf
+cd protobuf
+./autogen.sh
+./configure
+make -j
+sudo make install
+```
+
+#### macOS
+
+```bash
+# Assume that brew is installed, or you could install brew by:
+# /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+brew install protobuf
+```
+
+### Build Xline from source
+
+```bash
+# Assume that rust compile environment installed, such as cargo, etc.
+
+# clone source code
+git clone https://github.com/datenlord/Xline
+
+# compile Xline
+cd Xline
+cargo build --release
+
+# compile lock client for validation
+RUSTFLAGS="-C target-feature=+crt-static" cargo build --release --bin lock_client
```

Comment:
I think the `crt-static` feature is no longer needed as we only run it locally. Because on master the `lock_client` have to be move into docker so the feature is required.

Reply:
done